src/lib/x509/x509cert.cpp

```diff
@@ -691,10 +691,20 @@ std::string X509_Certificate::to_string() const {
    out << "Issued: " << this->not_before().readable_string() << "\n";
    out << "Expires: " << this->not_after().readable_string() << "\n";
 
+   try {
+      auto pubkey = this->subject_public_key();
+      out << "Public Key [" << pubkey->algo_name() << "-" << pubkey->key_length() << "]\n\n";
+      out << X509::PEM_encode(*pubkey) << "\n";
+   } catch(Decoding_Error&) {
+      const AlgorithmIdentifier& alg_id = this->subject_public_key_algo();
+      out << "Public Key Invalid! OID=" << alg_id.oid().to_formatted_string()
+          << " Hex=" << hex_encode(this->subject_public_key_bitstring()) << "\n";
+   }
```

Comment:
I think it would be helpful if this were to print the error message botan produced when decoding this key, no?

```suggestion
   } catch(const Decoding_Error& ex) {
      const AlgorithmIdentifier& alg_id = this->subject_public_key_algo();
      out << "Public Key Invalid! OID=" << alg_id.oid().to_formatted_string()
          << " Error=" << ex.what() << "\n";
          << " Hex=" << hex_encode(this->subject_public_key_bitstring()) << "\n";
   }
```